G-Research/fasttrackml
# Fix bug preventing getting/listing artifacts with space in the name

This PR improves on the work from #368 and #367 and brings the following changes:
* **Fix bug preventing getting/listing artifacts with space in the name**
* **Propagate context to artifact storage providers**
* Tests cleanup
* Remove unnecessary error check
* Unify style
* Improve comments
* Remove unnecessary else statement
